### ISSUE TITLE
Section on ZWNJ and ZWJ

### ISSUE THREAD

**Issue by behnam** [w3c/alreq#97]
Importing from ACTION-80 (https://www.w3.org/International/groups/arabic-layout/track/actions/80)

This will set the ground (specially, some definitions) for https://github.com/w3c/alreq/issues/91 .

**Comment by behnam** [comment]
I'm expanding section "2.4 Joining" to explain for use of ZWJ/ZWNJ. Doing this, I noticed this paragraph:

> Almost all the non-alphabetical characters are non-joining. The few exceptions will be discussed in this document.

I'm not sure what exceptions this sentence is talking about. @mostafah, can you elaborate on this?

Also, we have this:

> Please refer to The Unicode Standard Version 8.0, Section 9.2, for full explanation of Arabic cursive joining.

I assume we have many references to make, to various parts of the Unicode standard, as well as other resources. Should we put all these references in a standalone appendix and link to those entries? @r12a, what do you think?

**Comment by mostafah** [comment]
ZWJ and Tatweel are the exceptions, and I added that sentence for the sake of correctness, but we may want to omit the promise of discussing them later.

**Comment by behnam** [comment]
> ZWJ and Tatweel are the exceptions, and I added that sentence for the sake of correctness, but we may want to omit the promise of discussing them later.

Thanks, @mostafah. So, since I'm adding text here to explain ZWJ/ZWNJ usage here, I'll drop that line.

**Comment by behnam** [comment]
Regarding question about link to Unicode spec, see https://github.com/w3c/alreq/issues/103

**Comment by behnam** [comment]
Here's my draft for updating "2.4 Joining" to expand definitions for joining behavior, and add definition for ZWNJ and ZWJ:
https://docs.google.com/document/d/1AmQK5Nj37Fl0VJbPGRtlF9b4SKRSoCZWf61iGftEvZQ/edit (Open for suggest edits and comments)

**Comment by r12a** [comment]
I made some purely editorial suggestions in the google doc. Hopefully they will stick when i leave the page...

**Comment by behnam** [comment]
Regarding Joining Rule 3: from https://github.com/w3c/alreq/issues/91#issuecomment-275577609

<img width="435" alt="screen shot 2017-04-11 at 10 40 21 am" src="https://cloud.githubusercontent.com/assets/37169/24917603/51f26880-1ea3-11e7-9fca-f818c6b9de38.png">

**Comment by behnam** [comment]
U+0640 TATWEEL used as Joining Enforcer in Persian abbreviation:

<img width="372" alt="screen shot 2017-04-11 at 11 18 03 am" src="https://cloud.githubusercontent.com/assets/37169/24919244/a401a974-1ea8-11e7-9f7e-0c38f93e36cd.png">

**Comment by behnam** [comment]
I created a new issue for "Section on Hyphenation" (https://github.com/w3c/alreq/issues/108) as a follow-up from the work here.

"Section on Abbreviation" (#91) would be another follow up.

**Comment by behnam** [comment]
TODO after #110:
* Add the missing images.
* Section on connection types and broken (dangling) connections.

Follow up work:
* More on joining segments and context-based joining in section on Line Break.
* More on elongation and joining segments.
* List counters: cover closed and open joining segments.

Misc from https://www.microsoft.com/typography/OpenTypeDev/arabic/intro.htm
* Contextual substitutions
* Dotted-Circle for invalid diacritic logic
